code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -82,16 +84,62 @@
 	else if(user != src && istype(user,/mob/living/simple_animal/hostile/regalrat))
 		. += "<span class='warning'>Who is this foolish false king? This will not stand!</span>"
 
+/mob/living/simple_animal/hostile/regalrat/handle_environment(datum/gas_mixture/environment)
+	. = ..()
+	if(stat == DEAD || !environment || !environment.gases[/datum/gas/miasma])
+		return
+	var/miasma_percentage = environment.gases[/datum/gas/miasma][MOLES] / environment.total_moles()
+	if(miasma_percentage>=0.25)
+		heal_bodypart_damage(1)
+
 /mob/living/simple_animal/hostile/regalrat/AttackingTarget()
+	if (busy)
+		return
 	. = ..()
-	if(istype(target, /obj/item/food/cheesewedge))
+	if(istype(target, /obj/machinery/disposal))
+		src.visible_message("<span class='warning'>[src] starts rummaging through the [target].</span>","<span class='notice'>You rummage through the [target]...</span>")
+		busy = TRUE
```

Comment:
You could probably kill this entire variable by using `DOING_INTERACTION` instead.